src/MX/MegaMenu/Model/Menu/Item.php

```diff
@@ -54,32 +83,29 @@ public function getItemData($item)
     protected function getCategories($item)
     {
         if ($item['content_type'] === self::CONTENT_TYPE_CATEGORY) {
-            $objectManager = ObjectManager::getInstance();
-
             $contentCategory = explode('/', $item['content_category']);
-            $categoryId = $contentCategory[1];
-
-            /** @var CategoryFactory $categoryFactory */
-            $categoryFactory = $objectManager->create(CategoryFactory::class);
-
-            // Parent Category
-            /** @var Category $category */
-            $category = $categoryFactory->create()->load($categoryId);
-
-            $categories = [
-                'category' => $this->getCategoryData($category),
-                'children' => []
-            ];
-
-            // Subcategories
-            $subcategories = $category->getChildrenCategories();
-            if ($subcategories) {
-                foreach ($subcategories as $subcategory) {
-                    $categories['children'][] = $this->getCategoryData($subcategory);
+            if ($contentCategory && isset($contentCategory[self::OFFSET_CATEGORY_ID])) {
+                $categoryId = $contentCategory[self::OFFSET_CATEGORY_ID];
+
+                // Parent Category
+                /** @var Category $category */
+                $category = $this->categoryFactory->create()->load($categoryId);
+
+                $categories = [
+                    'category' => $this->getCategoryData($category),
+                    'children' => []
+                ];
+
+                // Subcategories
+                $subcategories = $category->getChildrenCategories();
+                if ($subcategories) {
```

Comment:
use !empty($subcategories) be more clear

Reply:
actually you can remove the if since the foreach will be fine with an empty array (will do 0 iteration)